src/accelerate/accelerator.py

```diff
@@ -755,7 +755,8 @@ def main_process_first(self):
         ...     print(f"This will be printed by process {accelerator.process_index}")
         ```
         """
-        yield self.state.main_process_first()
```

Comment:
Hey there, just a thought: you also could have written this as `yield from self.state.main_process_first()`.
I personally prefer your way of doing it, it feels a bit more explicit.

Reply:
Agreed, explicit is great here. Especially with something advanced such as this

Reply:
Hi @lebrice!

Interestingly, I think what you suggest doesn't work, because `State.main_process_first()` is not a generator but another context manager. BTW this is why `PartialState.main_process_first()` does work -- it delegates to a generator.

This is related to what I found very interesting in the issue as I mentioned above: it _feels like_ `yield` or at least `yield from` should work here, but it doesn't. I think this is why ChatGPT is confident it works :D It suggested the following example should work, which actually produces an error (you can remove the decorator from  `inner_context_manager()` to make it work).

```python
from contextlib import contextmanager

@contextmanager
def outer_context_manager():
    print("Entering outer context")
    yield
    print("Exiting outer context")

@contextmanager
def inner_context_manager():
    print("Entering inner context")
    yield
    print("Exiting inner context")

@contextmanager
def combined_context_manager():
    print("Entering combined context")
    yield from inner_context_manager()
    print("Exiting combined context")

with outer_context_manager():
    with combined_context_manager():
        print("Inside combined context")
```